src/soilbiogeochem/SoilBiogeochemDecompCascadeMIMICSMod.F90

```diff
@@ -0,0 +1,1107 @@
+
+                     o_scalar(c,j) = max(o2stress_unsat(c,j), mino2lim)
+                  end do
+               end do
+            else
+               o_scalar(bounds%begc:bounds%endc,1:nlevdecomp) = 1._r8
+            end if
+         else
+            o_scalar(bounds%begc:bounds%endc,1:nlevdecomp) = 1._r8
+         end if
+
+      end if
+
+      ! Term that reduces decomposition rate at depth
+      do j = 1, nlevdecomp
+         do fc = 1, num_soilc
+            c = filter_soilc(fc)
+            if (use_vertsoilc) then
+               ! Using fixed e-folding depth as in
+               ! SoilBiogeochemDecompCascadeBGCMod.F90
+               depth_scalar(c,j) = exp(-zsoi(j) / decomp_depth_efolding)
+            else
+               depth_scalar(c,j) = 1.0_r8
+            end if
+         end do
+      end do
+
+      ! TODO Move to the params files
+      fmet_p1 = 0.85_r8
+      fmet_p2 = 0.85_r8
+      fmet_p3 = 0.013_r8
+      fmet_p4 = 40.0_r8
+      fchem_r_p1 = 0.1_r8
+      fchem_r_p2 = -3.0_r8
+      fchem_k_p1 = 0.3_r8
+      fchem_k_p2 = -3.0_r8
+      tauMod_factor = 0.01_r8  ! was tauMod_denom = 100
+      tauMod_min = 0.8_r8
+      tauMod_max = 1.2_r8
+      tau_r_p1 = 5.2e-4_r8
+      tau_r_p2 = 0.3_r8  ! or 0.4?
+      tau_k_p1 = 2.4e-4_r8
+      tau_k_p2 = 0.1_r8
+      ko_r = 4.0_r8
+      ko_k = 4.0_r8
+      densdep = 1.0_r8
+      desorpQ10 = 1.1_r8
+      t_soi_ref = 25.0_r8  ! deg C
+
+      ! calculate rates for all litter and som pools
+      ! TODO Ok that I reversed order of do-loops?
+      do fc = 1,num_soilc
+         c = filter_soilc(fc)
+
+         ! Time-dependent params from Wieder et al. 2015 & testbed code
+         ! TODO STILL MISSING N-related stuff: DIN...
+
+         ! TODO Did I translate this testbed code correctly?
+         ! ligninNratioAvg(c) =  &
+         !    (ligninNratio(c,leaf) * (cleaf2met(c) + cleaf2str(c)) + &
+         !     ligninNratio(c,froot) * (croot2met(c) + croot2str(c)) + &
+         !     ligninNratio(c,wood) * (cwd2str(c))) / &
+         !    max(0.001, cleaf2met(c) + cleaf2str(c) + &
+         !               croot2met(c) + croot2str(c) + cwd2str(c))
+         ! Concerns:
+         ! - all-pft avg of ratios .ne. to ratios of the averages
+         ! so better to calculate all-pft avg ligninNratios in pftconMod
+         ! - Or should we even calculate them only for pfts that are present?
+         ! If so, we could do that just before call p2c and then call p2c only
+         ! once for the column-level ligninNratioAvg
+         ligninNratioAvg = ((sum(lf_flig(1:)) / size(lf_flig(1:))) * &
+                            (sum(lflitcn(1:) / size(lflitcn(1:)))) * &
+                            leafc_to_litter(c) + &
+           (sum(fr_flig(1:)) / size(fr_flig(1:))) * &
+           (sum(frootcn(1:)) / size(frootcn(1:))) * &
+           frootc_to_litter(c) + &
+           cwd_flig * (cwdc_col(c) / cwdn_col(c)) * m_deadstem_to_litter(c)) / &
```

Comment:
Prefer the `cwdc / cwdn` ratio or `deadwdcn` here? Maybe we expect their values to be the same?

Reply:
Do the ligninNratioAvg calculation at the p level and then call p2c for ligninNratioAvg.

Once this code is running, check for high-freq variations in ligninNratioAvg. We may want to replace the pool_to_litter terms with some long term mean.

Reply:
> Prefer the `cwdc / cwdn` ratio or `deadwdcn` here? Maybe we expect their values to be the same?

Prefer the ratio rather than the parameter.

Reply:
`m_deadstemc_to_litter` represents the input to cwd and should be replaced by the output from cwd

Reply:
You folks have me lost at the intro I need visual projection to see blueprint to evaluate. Thanks I'm still in grade school on this coding field but learning each day more and more.🌎🤗